modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/entity/Detector.java

```diff
@@ -76,4 +77,59 @@
         private Date dateLastUpdated;
     }
 
+    @Data
+    public static class TrainingMetaData {
+
+        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
+        @Field(type = FieldType.Date, format = DateFormat.custom, pattern = "yyyy-MM-dd HH:mm:ss")
+        private Date dateLastTrained;
+
+        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
+        @Field(type = FieldType.Date, format = DateFormat.custom, pattern = "yyyy-MM-dd HH:mm:ss")
+        private Date dateNextTraining;
```

Comment:
we could think of using the name dateTrainingNextRun

Reply:
naming updated as suggested